### DISCUSSION TITLE
What is the minimum hardware required to run this is in real time?

### DISCUSSION THREAD

**Discussion by sw82** [roboflow/supervision#1546] [Q&A]
Basically this is my question ;)

Will a Raspberry be sufficient to run this on in real-time?

**Comment by TheRootDaemon** [comment]
## Your Pi can run supervision, but the "real-time" part depends on your use case...

#### Hardware Performance Matrix
Platform | Real-Time Capable? | FPS Range | Verdict
-- | -- | -- | --
Raspberry Pi 4 (CPU only) | No | 0.5-2 FPS | Too slow for most use cases
Raspberry Pi + Coral TPU | Yes | 15-30+ FPS | Sweet spot with optimization
Jetson Nano | Yes | 10-25 FPS | Reliable low-cost GPU option
Desktop with GPU | Definitely | 60+ FPS | Best choice for development

#### It definitely will run in your Pi, because it is just a python script
#### So all comes down to what your real-time use case is ...

Some of the below are my thought on some of the use cases ..
- Security monitoring 1-5 FPS is fine
- Interactive apps: Need 15+ FPS
- Robotics: 30+ FPS required

And it also depends on how complex your model is in terms of datasets and computation.
I once tried running a Object detection model using a ESP32 and it was barely functional in real time.

#### If you find a way to optimize it to get higher FPS in your PI, then most of the people would want to know the secrets !
